src/ServerRequest.php

```diff
@@ -31,6 +31,7 @@ class ServerRequest implements ServerRequestInterface
     public function __construct()
     {
         $this->createDerivedAttributes();
+        $this->initHeaders();
```

Comment:
If you call `initHeaders()` in the constructor, you also need to call it in `reset()`.
However preferably call it when using the headers (`getHeaders()`, `addHeaders()`, etc).